corpus/pages/views.py

```diff
@@ -25,3 +27,17 @@ def about_us(request):
             "socieites": societies,
         },
     )
+
+
+def sig(request, sig_name):
+    try:
+        sig = SIG.objects.get(name=sig_name)
+    except SIG.DoesNotExist:
+        return redirect("index")
```

Comment:
For this, don't redirect to index. Lets show a 404 instead, as that is the appropriate response. I think there should be a function called `get_object_or_404`. Just look into that.

Reply:
Yeah there is, but if he's put in song, isn't index page a better option? Just wondering

Reply:
anyways, changed